packages/core/src/build/static/ids.ts

```diff
@@ -60,68 +61,135 @@ export const getIds = ({
   const tableIds: TableIds = {};
 
   // Build function IDs
-  for (const [indexingFunctionKey, { astHash, sources }] of Object.entries(
-    functionInputs,
-  )) {
-    // Find tables written to and read from?
-    const tableNames = tableAccess
-      .filter(
-        (t) =>
-          t.access === "write" && t.indexingFunctionKey === indexingFunctionKey,
-      )
-      .map((t) => t.table);
-
-    const tables = dedupe(tableNames).map(
-      (tableName) => tableInputs[tableName],
-    );
+  for (const indexingFunctionKey of Object.keys(functionInputs)) {
+    const { functions, tables } = resolveDependencies({
+      name: indexingFunctionKey,
+      type: "function",
+      tableAccess,
+    });
 
     functionIds[indexingFunctionKey] = crypto
       .createHash("sha256")
       .update(
         JSON.stringify({
-          indexingFunctionKey,
-          astHash,
-          sources,
+          name: indexingFunctionKey,
+          type: "function",
+
+          functions,
           tables,
+          functionInputs: functions.map((f) => functionInputs[f]),
+          tableInputs: tables.map((t) => tableInputs[t]),
         }),
       )
       .digest("hex");
   }
 
   // Build table IDs
-  for (const [tableName, { schema }] of Object.entries(tableInputs)) {
-    const functionKeys = tableAccess
-      .filter((t) => t.access === "write" && t.table === tableName)
-      .map((t) => t.indexingFunctionKey);
-
-    const functions = dedupe(functionKeys).map((key) => functionInputs[key]);
+  for (const tableName of Object.keys(tableInputs)) {
+    const { functions, tables } = resolveDependencies({
+      name: tableName,
+      type: "table",
+      tableAccess,
+    });
 
     tableIds[tableName] = crypto
       .createHash("sha256")
       .update(
         JSON.stringify({
-          tableName,
-          schema,
+          name: tableName,
+          type: "table",
+
           functions,
+          tables,
+          functionInputs: functions.map((f) => functionInputs[f]),
+          tableInputs: tables.map((t) => tableInputs[t]),
         }),
       )
       .digest("hex");
   }
 
+  console.log({
+    tableIds,
+    functionIds,
+  });
+
   return { tableIds, functionIds };
 };
 
+/**
+ * Deterministically resolve all dependencies of a given function or table.
+ */
+const resolveDependencies = ({
```

Comment:
Bit strange to me that this helper handles both functions and tables - seems like the logic is pretty different, might make sense to separate into two helpers?